Sources/PackageModel/UserToolchain.swift

```diff
@@ -321,7 +321,11 @@ public final class UserToolchain: Toolchain {
         destination: Destination,
         environment: EnvironmentVariables
     ) throws -> [String] {
-        let swiftCompilerFlags = destination.toolset.knownTools[.swiftCompiler]?.extraCLIOptions ?? []
+        var swiftCompilerFlags = destination.toolset.knownTools[.swiftCompiler]?.extraCLIOptions ?? []
+
+        if let linker = destination.toolset.knownTools[.linker]?.path {
+            swiftCompilerFlags += ["-use-ld=\(linker.pathString)"]
```

Comment:
cc @compnerd could this interfere with the `use-ld` check that's used to infer the librarian? Specifically, this uses an absolute path rather than just the name — so maybe `determineLibrarian` should just look at the last path component?

Reply:
Does `-use-ld=` support absolute paths?  If so, yes, this would need to have an associated change to the testing for the librarian check for Windows.  Additionally, I think that you should be using `linker._nativePathString(escaped: false)` here ... to ensure that we actually test with the proper path on windows, e.g. `-use-ld=C:\Program Files\Swift\Toolchain\0.0.0+Asserts\usr\bin\lld-link.exe`.  Note the space in the path and the use of `\`.

Reply:
Hmm it looks like passing a path to `-fuse-ld` is supported but not recommended: https://github.com/apple/llvm-project/blob/1604b03867ff8c0ece77a61d16e3f609e3cfcb3c/clang/lib/Driver/ToolChain.cpp#L820

The correct flag expected by the clang driver seems to be `--ld-path`. Will switch to that instead.

Reply:
This raises another question though: should we attempt to derive the linker flavor based on `linker.path`? Or should we just pass `--ld-path` and allow SDK vendors to specify the flavor via `extraCLIArguments` if they wish?